jax/_src/lax/lax.py

```diff
@@ -4535,7 +4535,7 @@ def _abstractify(x):
 
 def _check_user_dtype_supported(dtype, fun_name=None):
   # Avoid using `dtype in [...]` because of numpy dtype equality overloading.
-  if isinstance(dtype, type) and dtype in {bool, int, float, complex}:
+  if isinstance(dtype, type) and dtype in {bool, int, float, builtins.complex}:
```

Comment:
The issue is that the builtin `complex` has been clobbered by `def complex(...)` in this file.